pegasus_wrapper/resource_request.py

```diff
@@ -157,12 +157,8 @@ def from_parameters(params: Parameters) -> ResourceRequest:
             run_on_single_node=params.optional_string("run_on_single_node"),
         )
 
-    def unify(self, other: ResourceRequest) -> ResourceRequest:
-        if isinstance(other, SlurmResourceRequest):
-            partition = other.partition
-        else:
-            partition = self.partition
-
+    def unify(self, other: "SlurmResourceRequest") -> "SlurmResourceRequest":
+        partition = other.partition or self.partition
         return SlurmResourceRequest(
             partition=partition.name,
```

Comment:
Should line 161 just be moved here? I'm not sure there's a reason to pass just the 'name' of the partition along when the partition object will suffice?

Reply:
I thought about that, but we don't just need the partition object - we need the name from the partition. Therefore, we need to confirm that the partition object exists on either `ResourceRequest` before selecting one.